controllers/osbuildenvconfig_controller.go

```diff
@@ -490,20 +660,323 @@ func (r *OSBuildEnvConfigReconciler) generateComposerService(serviceName, portNa
+		OSBuildWorkerS3CABundleFile            string
+		OSBuildWorkerS3CABundleDir             string
+		OSBuildWorkerS3CABundleKey             string
+	}{
+		RPMRepoDistribution:                    workerRPMRepoDistribution,
+		OSBuildComposerTag:                     conf.GlobalConf.WorkerOSBuildComposerVersion,
+		OSBuildTag:                             conf.GlobalConf.WorkerOSBuildVersion,
+		RHCredentialsDir:                       workerRHCredentialsDir,
+		RHCredentialsUsernameKey:               workerRHCredentialsUsernameKey,
+		RHCredentialsPasswordKey:               workerRHCredentialsPasswordKey,
+		OSBuildWorkerCertsDir:                  workerOSBuildWorkerCertsDir,
+		ComposerWorkerAPIServiceName:           composerWorkerAPIServiceName,
+		OSBuildWorkerConfigDir:                 workerOSBuildWorkerConfigDir,
+		OSBuildWorkerConfigFile:                workerOSBuildWorkerConfigConfigMapKey,
+		OSBuildWorkerS3CredsFile:               workerOSBuildWorkerConfigS3CredentialsFile,
+		OSBuildWorkerS3CredsDir:                workerOSBuildWorkerS3CredsDir,
+		OSBuildWorkerS3CredsAccessKeyIDKey:     workerOSBuildWorkerS3CredsAccessKeyIDKey,
+		OSBuildWorkerS3CredsSecretAccessKeyKey: workerOSBuildWorkerS3CredsSecretAccessKeyKey,
+		OSBuildWorkerS3CABundleFile:            workerOSBuildWorkerConfigS3CABundleFile,
+		OSBuildWorkerS3CABundleDir:             workerOSBuildWorkerS3CABundleDir,
+		OSBuildWorkerS3CABundleKey:             workerOSBuildWorkerS3CABundleKey,
+	}
+	return r.ensureConfigMapForTemplateFileExists(ctx, workerConfigPlaybookConfigMapName, workerConfigPlaybookConfigMapKey, workerConfigPlaybookTemplateFile, workerConfigPlaybookParams, instance)
+}
+
+func (r *OSBuildEnvConfigReconciler) ensureWorkerConfigAnsibleConfigExists(ctx context.Context, instance *osbuildv1alpha1.OSBuildEnvConfig) (bool, error) {
+	return r.ensureConfigMapForTemplateFileExists(ctx, workerConfigAnsibleConfigConfigMapName, workerConfigAnsibleConfigConfigMapKey, workerConfigAnsibleConfigTemplateFile, nil, instance)
+}
+
+func (r *OSBuildEnvConfigReconciler) ensureWorkerConfigInventoryExists(ctx context.Context, instance *osbuildv1alpha1.OSBuildEnvConfig, workerName, workerAddress, workerUser string) (bool, error) {
+	workerConfigInventoryParams := struct {
+		Address string
+		User    string
+		SSHKey  string
+	}{
+		Address: workerAddress,
+		User:    workerUser,
+		SSHKey:  path.Join(workerConfigJobSSHKeyDir, workerSSHKeysPrivateKeyKey),
+	}
+	return r.ensureConfigMapForTemplateFileExists(ctx, fmt.Sprintf(workerConfigInventoryConfigMapNameFormat, workerName), workerConfigInventoryConfigMapKey, workerConfigInventoryTemplateFile, workerConfigInventoryParams, instance)
+}
+
+func (r *OSBuildEnvConfigReconciler) ensureOSBuildWorkerConfigExists(ctx context.Context, instance *osbuildv1alpha1.OSBuildEnvConfig) (bool, error) {
+	workerOSBuildWorkerConfigParams := struct {
+		CredentialsFile string
+		Bucket          string
+		GenericS3       *struct {
```

Comment:
again - I think it will be better to export is as a type

Reply:
I don't really see the benefit, but I made the change